components/org.wso2.carbon.identity.api.server.organization.user.sharing.management/org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/sharing/management/v1/impl/UsersApiServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.impl;
+
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.UsersApiService;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.core.UsersApiServiceCore;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.factories.UsersApiServiceCoreFactory;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserShareRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserShareWithAllRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserSharedOrganizationsResponse;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserSharedRolesResponse;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserUnshareRequestBody;
+import org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1.model.UserUnshareWithAllRequestBody;
+import org.wso2.carbon.identity.organization.management.organization.user.sharing.exception.UserShareMgtClientException;
+import org.wso2.carbon.identity.organization.management.organization.user.sharing.exception.UserShareMgtException;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.organization.user.sharing.management.common.constants.UserSharingMgtConstants.RESPONSE_DETAIL_USER_SHARE;
+import static org.wso2.carbon.identity.api.server.organization.user.sharing.management.common.constants.UserSharingMgtConstants.RESPONSE_DETAIL_USER_UNSHARE;
+import static org.wso2.carbon.identity.api.server.organization.user.sharing.management.common.constants.UserSharingMgtConstants.RESPONSE_STATUS_PROCESSING;
+
+/**
+ * Implementation of the user sharing management APIs.
+ */
+public class UsersApiServiceImpl implements UsersApiService {
+
+    private final UsersApiServiceCore usersApiServiceCore;
+
+    public UsersApiServiceImpl() {
+
+        try {
+            this.usersApiServiceCore = UsersApiServiceCoreFactory.getUsersApiServiceCore();
+        } catch (IllegalStateException e) {
+            throw new RuntimeException("Error occurred while initiating ServerUserStoreService.", e);
```

Comment:
```suggestion
            throw new RuntimeException("Error occurred while initiating ServerUserStoreService.", e);
```
ServerUserStoreService -> wrong service name

Reply:
Addressed in commit: [8dc1bd5](https://github.com/wso2/identity-api-server/pull/784/commits/8dc1bd596d0aadbc0d93294e3cb17ede5342c0ae)